src/applications/personalization/profile/mocks/server.js

```diff
@@ -55,6 +55,35 @@ const genericErrors = {
   error403,
 };
 
+const requestHistory = [];
+
+const logRequest = req => {
+  const { body, url, method, params, query } = req;
+  const historyEntry = {};
+  // only add variables to requestHistory if they are not empty
+  if (!_.isEmpty(params)) {
+    historyEntry.params = params;
+  }
+  if (!_.isEmpty(query)) {
+    historyEntry.query = query;
+  }
+
+  if (!_.isEmpty(body)) {
+    try {
+      historyEntry.body = JSON.parse(body);
+    } catch (e) {
+      historyEntry.body = body;
+    }
+  }
+
+  historyEntry.method = method;
+  historyEntry.url = url;
+
+  debug(JSON.stringify(historyEntry, null, 2));
+
+  requestHistory.push({ ...historyEntry, url, method });
+};
+
```

Comment:
I thought this was kinda cool. Essentially it creates a history of previous requests that can be used to do anything or just log out the request history of your mock server. I am using it to know whether 2 requests in the past was a status requests with a specific id, but I'm sure it could be useful for other things.